default.nix

```diff
@@ -0,0 +1,39 @@
+# TODO: Download rust version and targets via nix expression instead of rustup.
+
+let
+  pkgs = import <nixpkgs> { };
+  nodejs = pkgs.nodejs_latest;
+in pkgs.stdenv.mkDerivation {
+  name = "rustenv";
+  buildInputs = with pkgs; [
+    zlib.dev
+    nodejs
+    cmake
+    pkg-config
+    openssl.dev
+    protobuf
+
+    # Dependencies just used in makefile
```

Comment:
I don't see a `Makefile`... is this TBD?

Reply:
The `default.nix` is my rough working cut of a nix file for the whole project. It could be removed instead. The `Makefile` here is thus the projects root `Makefile`.